src/transformers/modeling_utils.py

```diff
@@ -4119,10 +4158,6 @@ def _find_mismatched_keys(
                 )
             raise RuntimeError(f"Error(s) in loading state_dict for {model.__class__.__name__}:\n\t{error_msg}")
 
-        if is_quantized:
```

Comment:
Why this has been removed?

Reply:
this part of code was originally meant to deal with loaded components that are not params or buffers. Now this is addressed in  `load_state_dict_into_meta_model()` [here](https://github.com/huggingface/transformers/pull/26037/files#diff-6b72b98c4c2dcfc6cc606843917733f5d858374fbc22a735ff483bbc0c1e63eaR774)

Reply:
I thought that having in the main function code for dealing with individual components (and adding 4bit stuff there) was not pretty.

In #26610 I take this idea further and [have a method update_mismatched_keys()](https://github.com/huggingface/transformers/pull/26610/files#diff-6b72b98c4c2dcfc6cc606843917733f5d858374fbc22a735ff483bbc0c1e63eaR3620) for the same purpose.